src/main/java/graphql/execution/AsyncExecutionStrategy.java

```diff
@@ -42,35 +42,24 @@ public CompletableFuture<ExecutionResult> execute(ExecutionContext executionCont
 
         ExecutionStrategyInstrumentationContext executionStrategyCtx = ExecutionStrategyInstrumentationContext.nonNullCtx(instrumentation.beginExecutionStrategy(instrumentationParameters, executionContext.getInstrumentationState()));
 
-        MergedSelectionSet fields = parameters.getFields();
-        List<String> fieldNames = fields.getKeys();
-        Async.CombinedBuilder<FieldValueInfo> futures = Async.ofExpectedSize(fields.size());
-        for (String fieldName : fieldNames) {
-            MergedField currentField = fields.getSubField(fieldName);
-
-            ResultPath fieldPath = parameters.getPath().segment(mkNameForPath(currentField));
-            ExecutionStrategyParameters newParameters = parameters
-                    .transform(builder -> builder.field(currentField).path(fieldPath).parent(parameters));
-
-            CompletableFuture<FieldValueInfo> future = resolveFieldWithInfo(executionContext, newParameters);
-            futures.add(future);
-        }
+        List<String> fieldNames = parameters.getFields().getKeys();
```

Comment:
the above is moved into the `getAsyncFieldValueInfo` method since its commonly used